spec/date.html

```diff
@@ -475,31 +475,31 @@ <h1>DifferenceDate ( _smaller_, _greater_, _largestUnit_ )</h1>
         1. Assert: _largestUnit_ is one of `"years"`, `"months"`, `"days"`, `"hours"`, `"minutes"`, or `"seconds"`.
         1. If _largestUnit_ is not `"years"` or `"months"`,
           1. Set _largestUnit_ to `"days"`.
-        1. Let _years_ be _greater_.[[Year]] &minus; _smaller_.[[Year]].
+        1. Let _years_ be _greater_.[[IsoYear]] &minus; _smaller_.[[IsoYear]].
         1. If _largestUnit_ is `"days"`, then
-          1. Let _days_ be ! ToDayOfYear(_greater_.[[Year]], _greater_.[[Month]], _greater_.[[Day]]) &minus; ! ToDayOfYear(_smaller_.[[Year]], _smaller_.[[Month]], _smaller_.[[Day]]).
+          1. Let _days_ be ! ToDayOfYear(_greater_.[[IsoYear]], _greater_.[[IsoMonth]], _greater_.[[IsoDay]]) &minus; ! ToDayOfYear(_smaller_.[[IsoYear]], _smaller_.[[IsoMonth]], _smaller_.[[IsoDay]]).
           1. Assert: _years_ ≥ 0.
           1. Repeat, while _years_ &gt; 0
-            1. Set _days_ to _days_ + ! DaysInYear(_smaller_.[[Year]] + _years_ &minus; 1).
+            1. Set _days_ to _days_ + ! DaysInYear(_smaller_.[[IsoYear]] + _years_ &minus; 1).
             1. Set _years_ to _years_ &minus; 1.
           1. Return the Record {
             [[Years]]: 0,
             [[Months]]: 0,
             [[Days]]: _days_
           }.
-        1. Let _months_ be _greater_.[[Month]] &minus; _smaller_.[[Month]].
-        1. Let _balanceResult_ be ? BalanceDurationDate(_years_, _months_, _smaller_.[[Year]], _smaller_.[[Month]], _smaller_.[[Day]]).
+        1. Let _months_ be _greater_.[[IsoMonth]] &minus; _smaller_.[[IsoMonth]].
+        1. Let _balanceResult_ be ? BalanceDurationDate(_years_, _months_, _smaller_.[[IsoYear]], _smaller_.[[IsoMonth]], _smaller_.[[IsoDay]]).
         1. Set _years_ to _balanceResult_.[[Years]].
         1. Set _months_ to _balanceResult_.[[Months]].
-        1. Let _days_ be ! ToDayOfYear(_greater_.[[Year]], _greater_.[[Month]], _greater_.[[Day]]) &minus; ! ToDayOfYear(_balanceResult_.[[Year]], _balanceResult_.[[Month]], _smaller_.[[Day]]).
+        1. Let _days_ be ! ToDayOfYear(_greater_.[[IsoYear]], _greater_.[[IsoMonth]], _greater_.[[IsoDay]]) &minus; ! ToDayOfYear(_balanceResult_.[[IsoYear]], _balanceResult_.[[Month]], _smaller_.[[IsoDay]]).
```

Comment:
```suggestion
        1. Let _days_ be ! ToDayOfYear(_greater_.[[IsoYear]], _greater_.[[IsoMonth]], _greater_.[[IsoDay]]) &minus; ! ToDayOfYear(_balanceResult_.[[IsoYear]], _balanceResult_.[[IsoMonth]], _smaller_.[[IsoDay]]).
```

Reply:
I ended up needing to actually change `_balanceResult_.[[IsoYear]]` back to `_balanceResult_.[[Year]]` here, since the result of the balance operation is still a record with [[Year]] and [[Month]] fields, not a date with [[ISOYear]] and [[ISOMonth]] internal slots.